src/base/settingsstorage.h

```diff
@@ -62,4 +62,34 @@ public slots:
     mutable QReadWriteLock m_lock;
 };
 
+template<typename T>
+class StorableValue
+{
+public:
+    explicit StorableValue(const std::string &keyName, const QVariant &defaultValue = QVariant())
```

Comment:
I still don't understand your concerns about `QVariant` and `T` conversion. If the type is not registered, `QVariant` triggers a nice `static_assert`. Could you explain, please, what is the advantage of your approach?